File: src/test/java/com/atlassian/migration/datacenter/api/fs/FileSystemMigrationEndpointTest.java
package com.atlassian.migration.datacenter.api.fs;

import com.atlassian.migration.datacenter.core.exceptions.FilesystemMigrationException;
import com.atlassian.migration.datacenter.core.exceptions.InvalidMigrationStageError;
import com.atlassian.migration.datacenter.spi.fs.FilesystemMigrationService;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import javax.ws.rs.core.Response;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.doThrow;

@ExtendWith(MockitoExtension.class)
class FileSystemMigrationEndpointTest {

    @Mock
    FilesystemMigrationService fsMigrationService;

    FileSystemMigrationEndpoint endpoint;

    @BeforeEach
    void setUp() {
        endpoint = new FileSystemMigrationEndpoint(fsMigrationService);
    }

    @Test
    void abortRunningMigrationShouldBeSuccessful() {
        final Response response = endpoint.abortFilesystemMigration();
        assertEquals(response.getStatus(), Response.Status.OK.getStatusCode());
    }

    @Test
    void throwConflictIfMigrationIsNotRunning() throws Exception {
        doThrow(new InvalidMigrationStageError("running")).when(fsMigrationService).abortMigration();

        final Response response = endpoint.abortFilesystemMigration();

        assertEquals(response.getStatus(), Response.Status.CONFLICT.getStatusCode());
    }

    @Test
    void returnBadResultIfThereIsFsMigrationError() throws Exception {
        doThrow(new FilesystemMigrationException("")).when(fsMigrationService).abortMigration();

        final Response response = endpoint.abortFilesystemMigration();

        assertEquals(response.getStatus(), Response.Status.BAD_REQUEST.getStatusCode());
    }
}

File: src/test/java/com/atlassian/migration/datacenter/api/fs/FileSystemMigrationEndpointTest.kt
package com.atlassian.migration.datacenter.api.fs

import com.atlassian.migration.datacenter.core.exceptions.InvalidMigrationStageError
import com.atlassian.migration.datacenter.spi.fs.FilesystemMigrationService
import io.mockk.MockKAnnotations
import io.mockk.every
import io.mockk.impl.annotations.InjectMockKs
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import io.mockk.just
import io.mockk.runs
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import javax.ws.rs.core.Response

@ExtendWith(MockKExtension::class)
internal class FileSystemMigrationEndpointTest {
    @MockK
    lateinit var fsMigrationService: FilesystemMigrationService

    @InjectMockKs
    lateinit var endpoint: FileSystemMigrationEndpoint

    @BeforeEach
    fun setUp() = MockKAnnotations.init(this)

    @Test
    fun abortRunningMigrationShouldBeSuccessful() {
        every { fsMigrationService.abortMigration() } just runs
        val response = endpoint.abortFilesystemMigration()
        Assertions.assertEquals(response.status, Response.Status.OK.statusCode)
    }

    @Test
    @Throws(Exception::class)
    fun throwConflictIfMigrationIsNotRunning() {
        every {fsMigrationService.abortMigration()} throws InvalidMigrationStageError("running")
        val response = endpoint.abortFilesystemMigration()
        Assertions.assertEquals(response.status, Response.Status.CONFLICT.statusCode)
    }
}